sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.JsonSchema/src/SchemaRegistryJsonSerializer.cs

```diff
@@ -0,0 +1,688 @@
+        private readonly string _groupName;
+        private readonly SchemaRegistryJsonSchemaGenerator _jsonSchemaGenerator;
+        private const string JsonMimeType = "application/json";
+        private const int CacheCapacity = 128;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SchemaRegistryJsonSerializer"/>. This constructor can only be used to create an
+        /// instance which will be used for deserialization. In order to serialize (or both serialize and deserialize) you will need to use
+        /// one of the constructors that have a <c>groupName</c> parameter.
+        /// </summary>
+        /// <param name="client">The <see cref="SchemaRegistryClient"/> instance to use for looking up schemas.</param>
+        /// <param name="jsonSchemaGenerator">The instance inherited from <see cref="SchemaRegistryJsonSchemaGenerator"/> to use to generate and validate JSON schemas.</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public SchemaRegistryJsonSerializer(SchemaRegistryClient client, SchemaRegistryJsonSchemaGenerator jsonSchemaGenerator)
+            : this(client, null, jsonSchemaGenerator)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SchemaRegistryJsonSerializer"/>. This constructor can be used to create an instance
+        /// that will work for both serialization and deserialization.
+        /// </summary>
+        /// <param name="client">The <see cref="SchemaRegistryClient"/> instance to use for looking up schemas.</param>
+        /// <param name="groupName">The Schema Registry group name that contains the schemas that will be used to serialize.</param>
+        /// <param name="jsonSchemaGenerator">The instance inherited from <see cref="SchemaRegistryJsonSchemaGenerator"/> to use to generate and validate JSON schemas.</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public SchemaRegistryJsonSerializer(SchemaRegistryClient client, string groupName, SchemaRegistryJsonSchemaGenerator jsonSchemaGenerator)
+        {
+            _client = client ?? throw new ArgumentNullException(nameof(client));
+            _groupName = groupName;
+            _jsonSchemaGenerator = jsonSchemaGenerator ?? throw new ArgumentNullException(nameof(jsonSchemaGenerator));
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SchemaRegistryJsonSerializer"/> class for mocking use in testing.
+        /// </summary>
+        /// <remarks>
+        /// This constructor exists only to support mocking. When used, class state is not fully initialized, and
+        /// will not function correctly; virtual members are meant to be mocked.
+        ///</remarks>
+        protected SchemaRegistryJsonSerializer()
+        {
+        }
+
+        private readonly LruCache<string, string> _idToSchemaMap = new(CacheCapacity);
```

Comment:
Are these heavy enough that we should consider lazy instantiation?

Reply:
I think in theory yes, but since all of the methods make use of both of these caches, I'm not sure if it would make too much of a difference.